### DISCUSSION TITLE
How to execute project?

### DISCUSSION THREAD

**Discussion by jongfeel** [wisdom08/dev-community#11] [Q&A]
README 문서가 없는 게 문제이기도 하고

소스 코드를 다운로드 받고 실행하기 위한 방법은 무엇일까요?

gradle 쓰는 것 같아 

``` 
gradlew bootRun
```

을 해봤는데 에러가 나는군요

![image](https://user-images.githubusercontent.com/17442457/189679008-ff289f2d-f803-4f75-b271-1a75df9489ee.png)

**Comment by wisdom08** [comment]
@jongfeel 
음.. 총제적으로 좀 문제가 많은 것 같네요 😨

우선 readme 파일을 지금 추가하고 프로젝트가 실행되지 않는 원인도 찾아보겠습니다. 

gradle 프로젝트가 맞고 JDK 버전이 17입니다. 

여쭤보고 싶은게 application-oauth.yaml 파일에 oauth 관련 id값과 secret값이 있고 이 파일을 깃이그노어 파일에 추가를 했는데요. 
멘토님께서 실행하려면 이 파일은 어떻게 처리를 해야하는 건가요? 

헤로쿠에 배포했을 때는 헤로쿠 사이트에서 환경 변수 설정에 추가를 했었습니다. 배포 전에는 어떻게 해야하는지 관련된 내용을 찾아보겠습니다.

**Reply by jongfeel** [reply]
application-oauth.yaml을 전달하는 쉬운 방법은 이메일로 파일을 저에게 전달하는 겁니다.
그 후에는 oauth secret 값을 어떻게 숨겨야 하는지에 대한 방법을 찾아 보는 거죠.
issue에 새로 등록하면 좋을 겁니다.

**Reply by wisdom08** [reply]
네 이메일로 우선 전달해드렸습니다.
issue 등록하고 해결해볼게요!

**Reply by jongfeel** [reply]
일단 `feature/#4-oauth` branch에서
application-oauth.yaml 파일을 src/main/resources에 두고
gradlew bootRun을 했을 때 실행이 되긴 했습니다...만

추가로 mysql이 설치되어 있지 않은 환경에서는 이 프로젝트가 동작하지 않는 것도 추가로 해결해야 할 것 같습니다.

그러므로 mysql 설치에 대한 문서와 생성해 둬야 하는 database 부터 해서 table 기타 추가로 생성했던 것들에 대한 script도 문서화가 필요할 것 같아요.
script도 지속적으로 update가 될 것이므로 github로 관리하면 좋을 것 같군요.

처음에 협업 할 때 이런 걸 잘 맞춰두고 해야 나중에 무슨 일(?)이 생겨도 대처가 가능할 수 있습니다.

**Reply by wisdom08** [reply]
아 다행입니다 ㅜㅠ 
mysql이 설치되어 있지 않은 환경에서는 이 프로젝트가 동작하지 않는 것도 고려해야하는군요. 
고려해야할 게 정말 많네요!
말씀해주셔서 감사합니다. 
이 부분도 readme에 추가해두겠습니다.
